node-src/lib/getFileHashes.ts

```diff
@@ -0,0 +1,72 @@
+import { close, open, read } from 'fs';
+import pLimit from 'p-limit';
+import { join } from 'path';
+import xxHashWasm, { XXHash, XXHashAPI } from 'xxhash-wasm';
+
+const hashFile = (buffer: Buffer, path: string, xxhash: XXHashAPI): Promise<string> => {
+  const BUFFER_SIZE = buffer.length;
+
+  return new Promise((resolve, reject) => {
+    const done = (fd: number, getResult: () => bigint) => {
+      let result: bigint = undefined;
+      close(fd, (closeErr) => {
+        if (closeErr) reject(closeErr);
+        else resolve(result.toString(16).padStart(16, '0'));
+      });
+      result = getResult();
+    };
+
+    const readIncremental = (fd: number, hash: XXHash<bigint>) => {
+      read(fd, buffer, null, BUFFER_SIZE, -1, (readErr, bytesRead) => {
+        if (readErr) {
+          return close(fd, () => reject(readErr));
+        }
+        if (bytesRead === BUFFER_SIZE) {
+          hash.update(buffer);
+          readIncremental(fd, hash);
+        } else {
+          hash.update(buffer.subarray(0, bytesRead));
+          done(fd, () => hash.digest());
+        }
+      });
+    };
+
+    open(path, 'r', (openErr, fd) => {
```

Comment:
Do we really need to use this oldschool node callback style of programming? Can we not import from `fs/promises` and make things a bit easier to understand?

Reply:
Perhaps it would be useful to leave a comment explaining _why_ this style was chosen? (I think Gert told me someone proved it to be faster than the alternative)